eigenpro/kernels/utils.py

```diff
@@ -0,0 +1,350 @@
+        return ohe.reshape(seq_shape + (max_seq_len, -1,))
+    elif not np.all(seqs <= 1) and (not np.any(np.isnan(seqs))):
+        # seqs is inds
+        logging.warning("I would write a new ohe func if you're ohe'ing inds")
+        ohe = seqs[..., None] == np.arange(alphabet_size)
+        return np.nan_to_num(ohe, 0)
+    else:
+        # seqs is already ohe
+        return seqs
+    
+def get_binarize(seqs, alphabet_name='dna', include_stop=False, fill_stop=False):
+    """ Represents sequences as binary ints: empty spots are 1,
+    the first letter is 10, second is 100, etc... . For easy
+    Hamming distance calculation.
+    
+    Parameters:
+    seqs: numpy array of strs or ints
+        First put through get_ohe, accepts all representations except binary.
+    alphabet_name: str
+    include_stop: bool
+    fill_stop: bool
+    
+    Returns:
+    bin_seqs: numpy array
+    """    
+    ohe = get_ohe(seqs, alphabet_name=alphabet_name)
+    alphabet_size = np.shape(ohe)[-1]
+    return np.dot(ohe, 2**(np.arange(alphabet_size)+1)-1).astype(int) + 1
+
+
+def get_inds(seqs, alphabet_name='dna', include_stop=False):
+    """ Represents sequences as ints: empty spots are nan,
+    the first letter is 0, second is 1, etc... . 
+    
+    Parameters:
+    seqs: numpy array of strs or ints
+        First put through get_ohe, accepts all representations except binary.
+    alphabet_name: str
+    include_stop: bool
+        Include stop as a letter, distinct from empty spots.
+    
+    Returns:
+    bin_seqs: numpy array
+    """
+    alphabet = alphabets_en[alphabet_name]
+    alphabet = alphabet[:len(alphabet) - 1 + include_stop]
+    alphabet_size = len(alphabet)
+    inds = np.dot(get_ohe(
+        seqs, alphabet_name=alphabet_name, include_stop=include_stop),
+                  np.arange(alphabet_size, dtype=int)+1).astype(float)
+    inds[inds == 0] = np.nan
+    return inds - 1
+
+
+def get_str(seqs, alphabet_name='dna', include_stop=False):
+    """ Get string representation of OHE sequences. 
+    
+    Parameters:
+    seqs: numpy array
+    alphabet_name: str, default = 'dna'
+    include_stop: bool, default = False
+        Whether to incldue the stop in the alphabet.
+        (Does the ohe have an extra row for stop?)
+        
+    Returns:
+    strs: numpy array
+        Has shape seqs.shape[:-2].
+    """
+    alphabet = alphabets_en[alphabet_name]
+    alphabet = alphabet[:len(alphabet) - 1 + include_stop]
+    seqs_flat = seqs.reshape((-1,) + np.shape(seqs)[-2:])
+    lens = get_lens(seqs_flat).astype(int)
+    strs = [''.join([alphabet[ohe][0] for ohe in seq[:len_].astype(bool)])
+            for seq, len_ in zip(seqs_flat, lens)]
+    return np.array(strs).reshape(np.shape(seqs)[:-2])
+
+
+def get_lens(seqs_ohe):
+    """ Get lengths of OHE sequences - numpy arrays or torch tensors. """
+    return seqs_ohe.sum(axis=-1).sum(axis=-1)
+
+
+################ Reformat axes of sequence representations #############
+
+
```

Comment:
can we remove these two blank lines?

Reply:
Sorry I got rid of the entire file, because now I'm using a pytorch implementation of imq